.github/actions/changelog-generator/generate_changelog.py

```
@@ -0,0 +1,318 @@
+    diff = get_pr_diff(base_branch)
+    commits = get_pr_commits(base_branch)
+    issues = get_linked_issues(repo, pr_details["body"], github_token)
+
+    prompt = build_prompt(pr_details, diff, commits, issues)
+    changelog_entry = call_gemini(prompt, api_key, model)
+    changelog_entry = changelog_entry.strip()
+    comment = f"```rst\n{changelog_entry}\n```"
+    try:
+        post_github_comment(repo, pr_number, comment, github_token)
+    except (HTTPError, URLError) as e:
+        raise RuntimeError(f"Failed to post comment: {e}") from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)